pkg/dataMover/data_mover.go

```diff
@@ -87,6 +89,15 @@ func NewDataMoverFromCluster(params map[string]interface{}, logger logrus.FieldL
 }
 
 func (this *DataMover) CopyToRepo(peID astrolabe.ProtectedEntityID) (astrolabe.ProtectedEntityID, error) {
+	backupRepository := builder.ForBackupRepository(utils.WithoutBackupRepository).Result()
```

Comment:
Could you change the call sites as well?, in upload controller check if the upload CR has a BR specified, if so call CopyToRepoWithBackupRepository